### ISSUE TITLE
Wrong estimation of label-based affine transformation 

### ISSUE THREAD

**Issue by jcohenadad** [neuropoly/spinalcordtoolbox#1364]
### Steps to Reproduce

data: 20170616_issue1364/data
~~~
sct_register_to_template -i t2_pad.nii.gz -s t2_seg_pad.nii.gz -l labels_pad.nii -r 0 -v 2
~~~

### State of spinalcordtoolbox
Spinal Cord Toolbox (jca_issue1362/2b06e7a14fd10013d16df9d4b8d9c5d5ffc66879)

### Working branch
https://github.com/neuropoly/spinalcordtoolbox/tree/jca_issue1362

### Additional Information

![getrigidtransformfromlandmarks_plot](https://user-images.githubusercontent.com/2482071/27239164-d295fef6-529d-11e7-9ade-4b7a113e51ac.png)

**Comment by jcohenadad** [comment]
This seems to be related to the straight space, which is not colinear with the template, because register_to_template restrict affine transformation to TxTySz by default, assuming the straight space is aligned with the template.
@benjamindeleener: how do you create the physical coordinate of the straightened image? Is it always colinear to the template I-S axis?

**Comment by benjamindeleener** [comment]
@jcohenadad The straight space is aligned with the input image (same orientation matrix). I remember we had reasons to do so, and we should look for issues before changing it.

**Comment by jcohenadad** [comment]
As far as I know, comparing the input and input_straight matrices, they are not the same. Example on the data from this issue:

input
~~~
qto_xyz:1      -0.703003  -0.000000  0.102426  86.414833
qto_xyz:2      0.002476  0.690439  1.039857  -142.372238
qto_xyz:3      0.012858  -0.132959  5.399834  -170.047150
qto_xyz:4      0.000000  0.000000  0.000000  1.000000
~~~

input_straight
~~~
qto_xyz:1      -0.999827  -0.000000  0.018623  31.275379
qto_xyz:2      0.003521  0.981958  0.189065  -89.683372
qto_xyz:3      0.018287  -0.189098  0.981788  -187.296066
qto_xyz:4      0.000000  0.000000  0.000000  1.000000
~~~

**Comment by jcohenadad** [comment]
corretion on my previous comment: the reason qform (top left 3x3) are different is because the input data is reinterpolated to 1mm, and then straightened. When I compare the curve/straight with the same resolution, the headers are indeed the same (except the last 3x1 column, which is normal).

One solution, in order to minimize risks of breaking, would be to have straighten_spinalcord  have a flag to ask for output qform aligned with the template. @benjamindeleener what do you think?

**Comment by benjamindeleener** [comment]
Indeed, that would be a good idea. I've implemented it into the branch `jca_issue1362`, as a parameter `-param template_orientation=1`.

Here is the result of the straightening applied during the registration process.
![screen shot 2017-06-17 at 09 18 23](https://user-images.githubusercontent.com/7757117/27253196-0eb9cc14-533e-11e7-9320-f919058e5ca2.png)

However, the registration to the template seem to fail for both ways for these data:
```
Split destination volume...
0/113..
Traceback (most recent call last):
  File "/Users/benjamindeleener/code/sct/scripts/sct_register_to_template.py", line 646, in <module>
    main()
  File "/Users/benjamindeleener/code/sct/scripts/sct_register_to_template.py", line 448, in main
    warp_forward_out, warp_inverse_out = register(src, dest, paramreg, param, str(i_step))
  File "/Users/benjamindeleener/code/sct/scripts/sct_register_multimodal.py", line 710, in register
    verbose=param.verbose)
  File "/Users/benjamindeleener/code/sct/scripts/msct_register.py", line 61, in register_slicewise
    register2d_centermassrot('src.nii', 'dest.nii', fname_warp=warp_forward_out, fname_warp_inv=warp_inverse_out, rot=1, poly=int(paramreg.poly), path_qc=path_qc, verbose=verbose, pca_eigenratio_th=float(paramreg.pca_eigenratio_th))
  File "/Users/benjamindeleener/code/sct/scripts/msct_register.py", line 234, in register2d_centermassrot
    alpha=0.5)
  File "/Users/benjamindeleener/code/sct/python/lib/python2.7/site-packages/matplotlib/pyplot.py", line 3251, in scatter
    edgecolors=edgecolors, data=data, **kwargs)
  File "/Users/benjamindeleener/code/sct/python/lib/python2.7/site-packages/matplotlib/__init__.py", line 1812, in inner
    return func(ax, *args, **kwargs)
  File "/Users/benjamindeleener/code/sct/python/lib/python2.7/site-packages/matplotlib/axes/_axes.py", line 3865, in scatter
    x, y, s, c = cbook.delete_masked_points(x, y, s, c)
  File "/Users/benjamindeleener/code/sct/python/lib/python2.7/site-packages/matplotlib/cbook.py", line 1846, in delete_masked_points
    raise ValueError("Masked arrays must be 1-D")
ValueError: Masked arrays must be 1-D
```

**Comment by jcohenadad** [comment]
yes i know, i'm working on it

**Comment by peristeri** [comment]
Solved with #1368
